packages/app-accounts/src/DownloadButton.tsx

```diff
@@ -0,0 +1,148 @@
+// Copyright 2017-2018 @polkadot/app-accounts authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { Button$Sizes } from '@polkadot/ui-app/Button/types';
+import { BareProps, I18nProps } from '@polkadot/ui-app/types';
+import { KeyringPair$Json } from '@polkadot/util-keyring/types';
+
+import React from 'react';
+import FileSaver from 'file-saver';
+import keyring from '@polkadot/ui-keyring/index';
+import isUndefined from '@polkadot/util/is/undefined';
+import Button from '@polkadot/ui-app/Button';
+
+import DownloadModal from './DownloadModal';
+import translate from './translate';
+
+type State = {
+  address: string,
+  password: string,
+  isPasswordModalOpen: boolean,
+  error?: React.ReactNode
+};
+
+type Props = I18nProps & BareProps & {
+  icon?: string,
+  isCircular?: boolean,
+  isPrimary?: boolean,
+  size?: Button$Sizes,
+  address: string
+};
+
+class DownloadButton extends React.PureComponent<Props, State> {
+  state: State;
+
+  constructor (props: Props) {
+    super(props);
+
+    this.state = this.emptyState();
+  }
+
+  handleDownloadAccount = (): void => {
+    const { address, password } = this.state;
+
+    if (!address) {
+      return;
+    }
+
+    try {
+      const json: KeyringPair$Json | void = keyring.backupAccount(address, password);
+
+      if (!isUndefined(json)) {
+        const blob = new Blob([JSON.stringify(json)], { type: 'text/plain;charset=utf-8' });
+
+        FileSaver.saveAs(blob, `${address}.json`);
+
+        this.hidePasswordModal();
+      } else {
+        this.setState({
+          error: 'Unable to obtain account from memory'
```

Comment:
Translatable strings, we never us strings without `t(<key>, { defaultValue: <value> })`. So here it would be -

```js
handleDownloadAccount = (): void => {
  const { t } = this.props;

...

this.setState({
  error: t('download.error.memory', {
    defaultValue: 'Unable to obtain account from memory'
});
...
}
```